### PR TITLE
Add custom settings managment (mocked settings)

### PR DESCRIPTION
**Pull request by MarianneLa**

### CODE REVIEW

File: src/views/Main/Head/Head.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import { Helmet } from 'react-helmet';
+
+const Head = ({ settings }) => (
+  <Helmet>
+    <title>{settings.title}</title>
+    {/* {settings.theme.style.map(link => <link key={link} rel="stylesheet" type="text/html" href={link} />)} */}
```
Comment (jrmi) [inline src/views/Main/Head/Head.js]: Why do you keep this ?

File: src/views/Main/Header/Header.js [hunk 2]
```diff
@@ -13,23 +13,24 @@ import NavBarItemDesktop from './NavBarItemDesktop';
...
 import infoSign from '../../../images/info-sign.svg';
 import logIn from '../../../images/log-in.svg';
 import logOut from '../../../images/log-out.svg';
+import defaultLogo from '../../../images/terravisu-logo.svg';
 
 import './styles.scss';
 
-const generateMenus = (authenticated, views) => {
+const generateMenus = (authenticated, views, settings = {}) => {
+  const hasCustomLogo  = settings.theme && settings.theme.logo && settings.theme.logo.length;
```
Comment (jrmi) [inline src/views/Main/Header/Header.js]: Do you need the last part about the length ?

File: src/views/Main/Header/__snapshots__/index.test.js.snap [hunk 3]
```diff
@@ -6,6 +6,16 @@ exports[`should render correctly 1`] = `
 >
   <div
     className="bp3-navbar navBar bp3-dark"
-  />
+  >
+    <div
```
Comment (jrmi) [inline src/views/Main/Header/__snapshots__/index.test.js.snap]: Why do we have extra elements ?

File: src/views/Main/Provider/SettingsProvider.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+const { Provider } = contextSettings;
+
+const getSettings =  async () => {
+  // MOCK
+  const response = mockedSettings;
+  return response;
+};
+
+export class SettingsProvider extends React.Component {
```
Comment (jrmi) [inline src/views/Main/Provider/SettingsProvider.js]: You should create a functional component here.

File: src/views/Main/settings_mock.json [hunk 5]
```diff
@@ -0,0 +1,8 @@
+{
+  "favicon": "<favicon>",
+  "title": "title",
+  "theme": { 
+    "logo": "<URL>",
+    "style": ["<URL>"]
+  }
+}
```
Comment (jrmi) [inline src/views/Main/settings_mock.json]: Missing newline

File: src/views/Main/settings_mock.json [hunk 6]
```diff
@@ -0,0 +1,8 @@
+{
```
Comment (jrmi) [inline src/views/Main/settings_mock.json]: File should be name `settings_mock.dist.json`.

Comment (jrmi) [reply src/views/Main/settings_mock.json]: And add a default version with default value to keep compatibility. Also add information about this file in readme.

File: src/views/Main/Header/Header.js (hunk 2 above)
Comment (MarianneLa) [reply src/views/Main/Header/Header.js]: In case the sting is empty, it returns the default logo

File: src/views/Main/Head/Head.js [hunk 7]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import { Helmet } from 'react-helmet';
+
+const Head = ({ settings = {} }) => {
```
Comment (jrmi) [inline src/views/Main/Head/Head.js]: You may spread theme here ?

```js

const Head = ({ settings: {theme:{}={}, title} = {} }) => {
```

File: src/views/Main/Header/Header.js [hunk 8]
```diff
@@ -13,23 +13,22 @@ import NavBarItemDesktop from './NavBarItemDesktop';
...
-import logo from '../../../images/terravisu-logo.svg';
 
 import infoSign from '../../../images/info-sign.svg';
 import logIn from '../../../images/log-in.svg';
 import logOut from '../../../images/log-out.svg';
 
 import './styles.scss';
 
-const generateMenus = (authenticated, views) => {
+const generateMenus = (authenticated, views, settings = {}) => {
```
Comment (jrmi) [inline src/views/Main/Header/Header.js]: Here also.

File: src/views/Main/Provider/SettingsProvider.js [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+
+import { contextSettings } from './context';
+
+const { Provider } = contextSettings;
+const defaultSettings = {
+  favicon: '/favicon.png',
+  title: 'terra-visu',
+  theme: {
+    logo: '/images/terravisu-logo.png',
+    style: ['<URL>'],
```
Comment (jrmi) [inline src/views/Main/Provider/SettingsProvider.js]: Default settings have no style.
```suggestion
    style: [],
```

File: src/views/Main/Provider/SettingsProvider.js [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+  },
+};
+const getSettings =  async SETTINGS => {
+  try {
+    const customSettings = await fetch(`/${SETTINGS}`);
+    return await customSettings.json();
+  } catch (e) {
+    console.error('settings.json is invalid. Please create a public/settings.json from public/settings.dist.json');
```
Comment (jrmi) [inline src/views/Main/Provider/SettingsProvider.js]:
```suggestion
    console.error('settings.json is missing. Please create a public/settings.json from public/settings.dist.json');
```

File: src/views/Main/Head/Head.js (hunk 7 above)
Comment (dtrucs) [reply src/views/Main/Head/Head.js]: He would probably write:
const Head = ({ settings: { theme: { style } ={}, title } = {} }) => {

File: src/views/Main/Header/Header.js (hunk 8 above)
Comment (MarianneLa) [reply src/views/Main/Header/Header.js]: I can't spread theme here.

File: src/views/Main/settings_mock.json (hunk 6 above)
Comment (jrmi) [reply src/views/Main/settings_mock.json]: up

File: src/views/Main/Header/Header.js (hunk 8 above)
Comment (jrmi) [reply src/views/Main/Header/Header.js]: Why not ?

File: src/views/Main/settings_mock.json (hunk 5 above)
Comment (jrmi) [reply src/views/Main/settings_mock.json]: up

File: src/views/Main/Header/Header.js (hunk 8 above)
Comment (jrmi) [reply src/views/Main/Header/Header.js]:
```js
const generateMenus = (authenticated, views, {theme:{}} = {}) => {
```

File: src/views/Main/Provider/SettingsProvider.js (hunk 9 above)
Comment (MarianneLa) [reply src/views/Main/Provider/SettingsProvider.js]: done

File: src/views/Main/Head/Head.js (hunk 7 above)
Comment (MarianneLa) [reply src/views/Main/Head/Head.js]: done

File: src/views/Main/settings_mock.json (hunk 5 above)
Comment (MarianneLa) [reply src/views/Main/settings_mock.json]: Is settings_mock.json still useful?

File: src/views/Main/settings_mock.json (hunk 6 above)
Comment (MarianneLa) [reply src/views/Main/settings_mock.json]: cf /public/settings.dist.json

File: src/views/Main/Head/Head.js (hunk 7 above)
Comment (MarianneLa) [reply src/views/Main/Head/Head.js]: done

File: src/views/Main/settings_mock.json (hunk 5 above)
Comment (jrmi) [reply src/views/Main/settings_mock.json]: yes for deployment as long as the settings backend module is not activated.

File: src/views/Main/Header/Header.js [hunk 11]
```diff
@@ -13,23 +13,22 @@ import NavBarItemDesktop from './NavBarItemDesktop';
...
-import logo from '../../../images/terravisu-logo.svg';
 
 import infoSign from '../../../images/info-sign.svg';
 import logIn from '../../../images/log-in.svg';
 import logOut from '../../../images/log-out.svg';
 
 import './styles.scss';
 
-const generateMenus = (authenticated, views) => {
+const generateMenus = (authenticated, views, { theme } = {}) => {
```
Comment (jrmi) [inline src/views/Main/Header/Header.js]: Spread again:
```js
const generateMenus = (authenticated, views, { theme: {logo = "" } = {} } = {}) => {
```

File: src/views/Main/Header/Header.js (hunk 8 above)
Comment (MarianneLa) [reply src/views/Main/Header/Header.js]: done

File: src/views/Main/Provider/SettingsProvider.js [hunk 12]
```diff
@@ -0,0 +1,52 @@
+import React, { useState, useEffect } from 'react';
+
+import withEnv from '../../../config/withEnv';
+
+import { contextSettings } from './context';
+
+const { Provider } = contextSettings;
+const defaultSettings = {
```
Comment (jrmi) [inline src/views/Main/Provider/SettingsProvider.js]: Constant should be uppercase.

File: public/settings.dist.json [hunk 13]
```diff
@@ -0,0 +1,8 @@
+{
+  "favicon": "/favicon.png",
+  "title": "mock_title",
+  "theme": { 
+    "logo": "/images/mocked-logo.png",
+    "style": []
+  }
+}
```
Comment (jrmi) [inline public/settings.dist.json]: Please keep easy to replace placeholders here:

```suggestion
{
  "favicon": "<FAVICON>",
  "title": "<TITLE>",
  "theme": { 
    "logo": "<LOGO_PATH>",
    "style": [<STYLE_PATH>]
  }
}
```

File: src/views/Main/Provider/SettingsProvider.js [hunk 14]
```diff
@@ -0,0 +1,52 @@
...
+  favicon: '/favicon.png',
+  title: 'terra-visu',
+  theme: {
+    logo: '/images/terravisu-logo.png',
+    style: [],
+  },
+};
+const getSettings =  async SETTINGS => {
```
Comment (jrmi) [inline src/views/Main/Provider/SettingsProvider.js]: Why `SETTINGS` in uppercase here ?
also naming suggestion `settings_path`

Comment (MarianneLa) [reply src/views/Main/Provider/SettingsProvider.js]: done

File: public/settings.dist.json (hunk 13 above)
Comment (MarianneLa) [reply public/settings.dist.json]: done

File: src/views/Main/Header/Header.js (hunk 11 above)
Comment (MarianneLa) [reply src/views/Main/Header/Header.js]: done

File: src/views/Main/Provider/SettingsProvider.js (hunk 12 above)
Comment (MarianneLa) [reply src/views/Main/Provider/SettingsProvider.js]: which const ?

File: src/views/Main/Provider/SettingsProvider.js [hunk 15]
```diff
@@ -0,0 +1,52 @@
...
+const { Provider } = contextSettings;
+const defaultSettings = {
+  favicon: '/favicon.png',
+  title: 'terra-visu',
+  theme: {
+    logo: '/images/terravisu-logo.png',
+    style: [],
+  },
+};
+const getSettings =  async settings_path => {
```
Comment (jrmi) [inline src/views/Main/Provider/SettingsProvider.js]: Sorry i have done a mistake: => `settingsPath`

File: src/views/Main/Header/__snapshots__/index.test.js.snap (hunk 3 above)
Comment (jrmi) [reply src/views/Main/Header/__snapshots__/index.test.js.snap]: up ?

Comment (MarianneLa) [reply src/views/Main/Header/__snapshots__/index.test.js.snap]: done

File: src/views/Main/Provider/SettingsProvider.js (hunk 12 above)
Comment (MarianneLa) [reply src/views/Main/Provider/SettingsProvider.js]: done

**Pull request merged**